### PR TITLE
Analyze dynamic names for static property and const fetches

### PR DESCRIPTION
**Pull request by edsrzf**
While trying to use PHP 8.3's new dynamic class constant syntax, I found that the dynamic constant name wasn't being analyzed. Then I tried to see what dynamic static properties were doing and noticed that they had the same problem!

This fixes both issues.

### CODE REVIEW

File: src/Psalm/Internal/Analyzer/Statements/Expression/ClassConstAnalyzer.php [hunk 1]
```diff
@@ -206,7 +206,15 @@ public static function analyzeFetch(
             }
 
             if (!$stmt->name instanceof PhpParser\Node\Identifier) {
-                return true;
+                $was_inside_general_use = $context->inside_general_use;
+
+                $context->inside_general_use = true;
+
+                $ret = ExpressionAnalyzer::analyze($statements_analyzer, $stmt->name, $context);
```
Comment (weirdan) [inline src/Psalm/Internal/Analyzer/Statements/Expression/ClassConstAnalyzer.php:213]: I suppose it should be version guarded, to make sure we don't allow this syntax when target version is below 8.3

Comment (weirdan) [reply src/Psalm/Internal/Analyzer/Statements/Expression/ClassConstAnalyzer.php:213]: Also it's possible to access enum cases in the same way in 8.3, but it doesn't seem to be handled.

**Review by weirdan — CHANGES_REQUESTED** [review]

File: src/Psalm/Internal/Analyzer/Statements/Expression/ClassConstAnalyzer.php (hunk 1 above)
Comment (edsrzf) [reply src/Psalm/Internal/Analyzer/Statements/Expression/ClassConstAnalyzer.php:213]: Ah, I thought the version check would be handled by the parser, but apparently not. What behavior would you expect on a lower version than 8.3? `ParseError`?

Comment (weirdan) [reply src/Psalm/Internal/Analyzer/Statements/Expression/ClassConstAnalyzer.php:213]: `ParseError` sounds good, as that's what PHP emits: https://3v4l.org/DfCB2

**Review by weirdan — APPROVED** [review]

**Comment by weirdan** [comment]
Thanks!

**Pull request merged**